armadillo/src/main/java/at/favre/lib/armadillo/SecureSharedPreferences.java

```diff
@@ -253,6 +256,45 @@ public void unregisterOnSharedPreferenceChangeListener(OnSharedPreferenceChangeL
         sharedPreferences.unregisterOnSharedPreferenceChangeListener(onSharedPreferenceChangeListener);
     }
 
+    private final List<SharedPreferenceChangeListenerWrapper> securePreferenceListeners = new LinkedList<>();
+
+    public void registerOnSecurePreferenceChangeListener(@NonNull OnSecurePreferenceChangeListener listener) {
+        synchronized (securePreferenceListeners) {
+            ListIterator<SharedPreferenceChangeListenerWrapper> iterator = securePreferenceListeners.listIterator();
+            boolean found = false;
+            while (iterator.hasNext()) {
+                SharedPreferenceChangeListenerWrapper listenerWrapper = iterator.next();
+                OnSecurePreferenceChangeListener wrapped = listenerWrapper.getWrapped();
+                if (wrapped == null) {
+                    unregisterOnSharedPreferenceChangeListener(listenerWrapper);
+                    iterator.remove();
+                } else if (wrapped == listener) {
+                    found = true; // we let the loop continue, so that we may cleanup some more references
```

Comment:
Why add the complexity of checking if the listener reference is already added? The default implementation also doesn't care about this. I get stable behaviour, but adding the same reference twice would be a naiive developer error. See: https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/android/app/SharedPreferencesImpl.java

Reply:
Here as I was also doing some cleanup for released references I though that it wouldn't do harm to also add the safety check.

I can remove it, and just asume developer would do the correct thing. Do you recommend to also remove weak references cleanup loop?